In [1]:
import pandas as pd
import numpy as np
import string
import spacy

import matplotlib.pyplot as plt
from matplotlib.pyplot import imread
from wordcloud import wordcloud
%matplotlib inline

In [2]:
data = pd.read_csv("C:\\Users\\RUSHIKESH\\Downloads\\SalaryData_Train.csv")
data

age          workclass    education  educationno        maritalstatus  \
0       39          State-gov    Bachelors           13        Never-married   
1       50   Self-emp-not-inc    Bachelors           13   Married-civ-spouse   
2       38            Private      HS-grad            9             Divorced   
3       53            Private         11th            7   Married-civ-spouse   
4       28            Private    Bachelors           13   Married-civ-spouse   
...    ...                ...          ...          ...                  ...   
30156   27            Private   Assoc-acdm           12   Married-civ-spouse   
30157   40            Private      HS-grad            9   Married-civ-spouse   
30158   58            Private      HS-grad            9              Widowed   
30159   22            Private      HS-grad            9        Never-married   
30160   52       Self-emp-inc      HS-grad            9   Married-civ-spouse   

               occupation    relationship    race      sex  capitalgain  \
0            Adm-clerical   Not-in-family   White     Male         2174   
1         Exec-managerial         Husband   White     Male            0   
2       Handlers-cleaners   Not-in-family   White     Male            0   
3       Handlers-cleaners         Husband   Black     Male            0   
4          Prof-specialty            Wife   Black   Female            0   
...                   ...             ...     ...      ...          ...   
30156        Tech-support            Wife   White   Female            0   
30157   Machine-op-inspct         Husband   White     Male            0   
30158        Adm-clerical       Unmarried   White   Female            0   
30159        Adm-clerical       Own-child   White     Male            0   
30160     Exec-managerial            Wife   White   Female        15024   

       capitalloss  hoursperweek          native  Salary  
0                0            40   United-States   <=50K  
1                0            13   United-States   <=50K  
2                0            40   United-States   <=50K  
3                0            40   United-States   <=50K  
4                0            40            Cuba   <=50K  
...            ...           ...             ...     ...  
30156            0            38   United-States   <=50K  
30157            0            40   United-States    >50K  
30158            0            40   United-States   <=50K  
30159            0            20   United-States   <=50K  
30160            0            40   United-States    >50K  

[30161 rows x 14 columns]

In [6]:
data['Salary'].value_counts()

 <=50K    22653
 >50K      7508
Name: Salary, dtype: int64

In [7]:
# Lets do label-Encoding :-

In [9]:
from sklearn.preprocessing import LabelEncoder
label_encoder = LabelEncoder()

In [10]:
data['workclass'] = label_encoder.fit_transform(data.workclass)

In [12]:
data['education'] = label_encoder.fit_transform(data.education)

In [13]:
data['maritalstatus'] = label_encoder.fit_transform(data.maritalstatus)

In [14]:
data['occupation'] = label_encoder.fit_transform(data.occupation)

In [15]:
data['relationship'] = label_encoder.fit_transform(data.relationship)

In [16]:
data['race'] = label_encoder.fit_transform(data.race)

In [17]:
data['sex'] = label_encoder.fit_transform(data.sex)

In [18]:
data['native'] = label_encoder.fit_transform(data.native)

In [34]:
data

age  workclass  education  educationno  maritalstatus  occupation  \
0       39          5          9           13              4           0   
1       50          4          9           13              2           3   
2       38          2         11            9              0           5   
3       53          2          1            7              2           5   
4       28          2          9           13              2           9   
...    ...        ...        ...          ...            ...         ...   
30156   27          2          7           12              2          12   
30157   40          2         11            9              2           6   
30158   58          2         11            9              6           0   
30159   22          2         11            9              4           0   
30160   52          3         11            9              2           3   

       relationship  race  sex  capitalgain  capitalloss  hoursperweek  \
0                 1     4    1         2174            0            40   
1                 0     4    1            0            0            13   
2                 1     4    1            0            0            40   
3                 0     2    1            0            0            40   
4                 5     2    0            0            0            40   
...             ...   ...  ...          ...          ...           ...   
30156             5     4    0            0            0            38   
30157             0     4    1            0            0            40   
30158             4     4    0            0            0            40   
30159             3     4    1            0            0            20   
30160             5     4    0        15024            0            40   

       native  Salary  
0          37   <=50K  
1          37   <=50K  
2          37   <=50K  
3          37   <=50K  
4           4   <=50K  
...       ...     ...  
30156      37   <=50K  
30157      37    >50K  
30158      37   <=50K  
30159      37   <=50K  
30160      37    >50K  

[30161 rows x 14 columns]

In [35]:
# Lets make our data scale free :-

In [67]:
def std_data(i):
    result = (i-i.min())/(i.max()-i.min())
    return result

In [68]:
std_data = std_data(data.iloc[:,:-1])

In [69]:
#std_data['Salary']=data.Salary

In [70]:
std_data

age  workclass  education  educationno  maritalstatus  occupation  \
0      0.301370   0.833333   0.600000     0.800000       0.666667    0.000000   
1      0.452055   0.666667   0.600000     0.800000       0.333333    0.230769   
2      0.287671   0.333333   0.733333     0.533333       0.000000    0.384615   
3      0.493151   0.333333   0.066667     0.400000       0.333333    0.384615   
4      0.150685   0.333333   0.600000     0.800000       0.333333    0.692308   
...         ...        ...        ...          ...            ...         ...   
30156  0.136986   0.333333   0.466667     0.733333       0.333333    0.923077   
30157  0.315068   0.333333   0.733333     0.533333       0.333333    0.461538   
30158  0.561644   0.333333   0.733333     0.533333       1.000000    0.000000   
30159  0.068493   0.333333   0.733333     0.533333       0.666667    0.000000   
30160  0.479452   0.500000   0.733333     0.533333       0.333333    0.230769   

       relationship  race  sex  capitalgain  capitalloss  hoursperweek  \
0               0.2   1.0  1.0     0.021740          0.0      0.397959   
1               0.0   1.0  1.0     0.000000          0.0      0.122449   
2               0.2   1.0  1.0     0.000000          0.0      0.397959   
3               0.0   0.5  1.0     0.000000          0.0      0.397959   
4               1.0   0.5  0.0     0.000000          0.0      0.397959   
...             ...   ...  ...          ...          ...           ...   
30156           1.0   1.0  0.0     0.000000          0.0      0.377551   
30157           0.0   1.0  1.0     0.000000          0.0      0.397959   
30158           0.8   1.0  0.0     0.000000          0.0      0.397959   
30159           0.6   1.0  1.0     0.000000          0.0      0.193878   
30160           1.0   1.0  0.0     0.150242          0.0      0.397959   

         native  
0      0.948718  
1      0.948718  
2      0.948718  
3      0.948718  
4      0.102564  
...         ...  
30156  0.948718  
30157  0.948718  
30158  0.948718  
30159  0.948718  
30160  0.948718  

[30161 rows x 13 columns]

In [40]:
from sklearn.naive_bayes import GaussianNB
classifier = GaussianNB()


In [42]:
classifier.fit(std_data, data.Salary)


GaussianNB()

In [44]:
prediction = classifier.predict(std_data)
prediction

array([' <=50K', ' <=50K', ' <=50K', ..., ' <=50K', ' <=50K', ' >50K'],
      dtype='<U6')

In [47]:
from sklearn.metrics import accuracy_score


In [48]:
Accuracy = accuracy_score(prediction, data.Salary)

Accuracy

0.797586286926826

In [49]:
# for Testing data :-

In [51]:
test_data = pd.read_csv("C:\\Users\\RUSHIKESH\\Downloads\\SalaryData_Test.csv")
test_data

age      workclass      education  educationno        maritalstatus  \
0       25        Private           11th            7        Never-married   
1       38        Private        HS-grad            9   Married-civ-spouse   
2       28      Local-gov     Assoc-acdm           12   Married-civ-spouse   
3       44        Private   Some-college           10   Married-civ-spouse   
4       34        Private           10th            6        Never-married   
...    ...            ...            ...          ...                  ...   
15055   33        Private      Bachelors           13        Never-married   
15056   39        Private      Bachelors           13             Divorced   
15057   38        Private      Bachelors           13   Married-civ-spouse   
15058   44        Private      Bachelors           13             Divorced   
15059   35   Self-emp-inc      Bachelors           13   Married-civ-spouse   

               occupation    relationship                 race      sex  \
0       Machine-op-inspct       Own-child                Black     Male   
1         Farming-fishing         Husband                White     Male   
2         Protective-serv         Husband                White     Male   
3       Machine-op-inspct         Husband                Black     Male   
4           Other-service   Not-in-family                White     Male   
...                   ...             ...                  ...      ...   
15055      Prof-specialty       Own-child                White     Male   
15056      Prof-specialty   Not-in-family                White   Female   
15057      Prof-specialty         Husband                White     Male   
15058        Adm-clerical       Own-child   Asian-Pac-Islander     Male   
15059     Exec-managerial         Husband                White     Male   

       capitalgain  capitalloss  hoursperweek          native  Salary  
0                0            0            40   United-States   <=50K  
1                0            0            50   United-States   <=50K  
2                0            0            40   United-States    >50K  
3             7688            0            40   United-States    >50K  
4                0            0            30   United-States   <=50K  
...            ...          ...           ...             ...     ...  
15055            0            0            40   United-States   <=50K  
15056            0            0            36   United-States   <=50K  
15057            0            0            50   United-States   <=50K  
15058         5455            0            40   United-States   <=50K  
15059            0            0            60   United-States    >50K  

[15060 rows x 14 columns]

In [56]:
test_data['workclass'] = label_encoder.fit_transform(test_data.workclass)
test_data['education'] = label_encoder.fit_transform(test_data.education)
test_data['maritalstatus'] = label_encoder.fit_transform(test_data.maritalstatus)
test_data['occupation'] = label_encoder.fit_transform(test_data.occupation)
test_data['relationship'] = label_encoder.fit_transform(test_data.relationship)
test_data['race'] = label_encoder.fit_transform(test_data.race)
test_data['sex'] = label_encoder.fit_transform(test_data.sex)
test_data['native'] = label_encoder.fit_transform(test_data.native)


In [76]:
# make our test data scale free

In [74]:
std_test_data = std_data(test_data.iloc[:,:-1])

In [75]:
std_test_data

age  workclass  education  educationno  maritalstatus  occupation  \
0      0.109589   0.333333   0.066667     0.400000       0.666667    0.461538   
1      0.287671   0.333333   0.733333     0.533333       0.333333    0.307692   
2      0.150685   0.166667   0.466667     0.733333       0.333333    0.769231   
3      0.369863   0.333333   1.000000     0.600000       0.333333    0.461538   
4      0.232877   0.333333   0.000000     0.333333       0.666667    0.538462   
...         ...        ...        ...          ...            ...         ...   
15055  0.219178   0.333333   0.600000     0.800000       0.666667    0.692308   
15056  0.301370   0.333333   0.600000     0.800000       0.000000    0.692308   
15057  0.287671   0.333333   0.600000     0.800000       0.333333    0.692308   
15058  0.369863   0.333333   0.600000     0.800000       0.000000    0.000000   
15059  0.246575   0.500000   0.600000     0.800000       0.333333    0.230769   

       relationship  race  sex  capitalgain  capitalloss  hoursperweek  \
0               0.6  0.50  1.0     0.000000          0.0      0.397959   
1               0.0  1.00  1.0     0.000000          0.0      0.500000   
2               0.0  1.00  1.0     0.000000          0.0      0.397959   
3               0.0  0.50  1.0     0.076881          0.0      0.397959   
4               0.2  1.00  1.0     0.000000          0.0      0.295918   
...             ...   ...  ...          ...          ...           ...   
15055           0.6  1.00  1.0     0.000000          0.0      0.397959   
15056           0.2  1.00  0.0     0.000000          0.0      0.357143   
15057           0.0  1.00  1.0     0.000000          0.0      0.500000   
15058           0.6  0.25  1.0     0.054551          0.0      0.397959   
15059           0.0  1.00  1.0     0.000000          0.0      0.602041   

         native  
0      0.948718  
1      0.948718  
2      0.948718  
3      0.948718  
4      0.948718  
...         ...  
15055  0.948718  
15056  0.948718  
15057  0.948718  
15058  0.948718  
15059  0.948718  

[15060 rows x 13 columns]

In [77]:
test_prediction = classifier.predict(std_test_data)

In [79]:
Accuracy = accuracy_score(test_prediction, test_data.Salary)
Accuracy


0.796812749003984

#### we are getting the accuracy for testing data is 79.68 %.